### PR TITLE
RCHAIN-2307 Initial code reorganisation.

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
Initial code reorganisation for https://github.com/rchain/rchain/pull/2097

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
RCHAIN-2307

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 1]
```diff
@@ -20,10 +20,33 @@ import coop.rchain.rspace.{Blake2b256Hash, ReplayException}
 import scala.collection.immutable
 import scala.concurrent.ExecutionContext
 
-class RuntimeManager[F[_]: Concurrent] private (
+abstract class RuntimeManager[F[_]: Concurrent] {
```
Comment (ArturGajowy) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: why not a trait?

File: casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala [hunk 2]
```diff
@@ -106,38 +110,43 @@ object BlockApproverProtocol {
...
-              d =>
-                genesisBlessedTerms.contains(d.deploy.term.get) && genesisBlessedDeploys
-                  .exists(dd => deployDataEq.eqv(dd, d.deploy.raw.get))
-            )
-            .either(())
-            .or("Candidate deploys do not match expected deploys.")
-      _ <- (blockDeploys.size == genesisBlessedContracts.size)
-            .either(())
-            .or("Mismatch between number of candidate deploys and expected number of deploys.")
+      r                         <- validate
```
Comment (ArturGajowy) [inline casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala]: A blank line above and below the extracted method would greatly help with readability. Also, this should be a separate commit saying "Extract `validate` method". I'd then maybe refrain from checking out the PR and just do the review in github :)

Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala]: > should be a separate commit saying "Extract validate method"

Notice I only know that because I did a diff in IDEA :)

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala (hunk 1 above)
Comment (dzajkowski) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: I like the `[F[_]: Concurrent]` notation and traits don't accept it.
AFAIK an abstract class does not differ from a trait in this context.

File: casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala (hunk 2 above)
Comment (dzajkowski) [reply casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala]: would you prefer it being a separate top level method?

**Comment by dzajkowski** [comment]
bors try

File: casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala (hunk 2 above)
Comment (ArturGajowy) [reply casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala]: no preference regarding that :)

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala (hunk 1 above)
Comment (goral09) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: Out of curiosity - if there exists only one implementation of it why bother turning it into an abstract class?

Comment (dzajkowski) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: as stated in the blurb: it's for https://github.com/rchain/rchain/pull/2097

**Comment by dzajkowski** [comment]
bors try
